src/PlanView/GeoFenceEditor.qml

```diff
@@ -199,17 +200,70 @@ QGCFlickable {
                         }
 
                         QGCLabel {
-                            text:               qsTr("Delete")
+                            text:               qsTr("Max Altitude\nAMSL")
                             Layout.column:      2
                             Layout.alignment:   Qt.AlignHCenter
                         }
 
+                        Repeater {
+                            model: myGeoFenceController.polygons
+
+                            QGCTextField {
+                                text: object.maxAltitude == 0 ? "Disabled" : object.maxAltitude
+                                textColor: acceptableInput ? "black" : "red"
+                                unitsLabel: "m"
+                                showUnits: true
+                                validator: RegExpValidator{ regExp: /^[0-9]{1,5}|disabled$/i }
```

Comment:
According to https://doc.qt.io/qt-5/qml-qtquick-regexpvalidator.html#details, `RegExpValidator` is deprecated and should be replaced by `RegularExpressionValidator`

Reply:
I think RegularExpressionValidator is not yet supported by QTextField as of qt 5.15, but will require update to some version of 6.x first.